unity-client/Assets/Scripts/MainScripts/DCL/Environment/Environment.cs

```diff
@@ -0,0 +1,97 @@
+﻿using System.ComponentModel;
+using DCL.Controllers;
+using UnityEngine;
+using UnityEngine.SceneManagement;
+
+namespace DCL
+{
+    public class Environment
+    {
+        public static Model i = new Model();
+        private static System.Func<MessagingContext> messagingBuilder;
+        private static System.Func<PlatformContext> platformBuilder;
+        private static System.Func<WorldRuntimeContext> worldRuntimeBuilder;
+
+        public static void SetupWithDefaults()
+        {
+            Setup(
+                messagingBuilder: MessagingContextFactory.CreateDefault,
+                platformBuilder: PlatformContextFactory.CreateDefault,
+                worldRuntimeBuilder: WorldRuntimeContextFactory.CreateDefault
+            );
+        }
+
+        public static void Setup(System.Func<MessagingContext> messagingBuilder,
+            System.Func<PlatformContext> platformBuilder,
+            System.Func<WorldRuntimeContext> worldRuntimeBuilder)
+        {
+            Environment.messagingBuilder = messagingBuilder;
+            Environment.platformBuilder = platformBuilder;
+            Environment.worldRuntimeBuilder = worldRuntimeBuilder;
+            Setup();
+        }
+
+        public static void Setup()
+        {
+            i = new Model(messagingBuilder, platformBuilder, worldRuntimeBuilder);
+            Initialize();
+        }
+
+        public static void Initialize()
+        {
+            Model model = i;
+
+            //TODO(Brian): We can move to a RAII scheme + promises later to make this
+            //             more scalable.
+            model.world.sceneController.Initialize();
+            model.messaging.messagingControllersManager.Initialize(i.world.sceneController);
+            model.world.pointerEventsController.Initialize();
+            model.platform.memoryManager.Initialize();
+            model.world.worldState.Initialize();
+            model.world.worldBlockersController.InitializeWithDefaultDependencies(
+                model.world.worldState,
+                DCLCharacterController.i.characterPosition);
+            model.platform.parcelScenesCleaner.Start();
+            model.platform.cullingController.Start();
+            model.world.sceneBoundsChecker.Start();
+        }
```

Comment:
In this method, each subsystem dependencies are wired together. I don't like this approach at all because it can turn to a big ball of mud pretty easily. 

For now, this is necessary because a system can reference another that doesn't exist yet. 

An approach to keep using constructors and get rid of these Start/Initialize/etc methods is to have promises that ensure that a subsystem will be available at a later time, and pass them through each subsystem constructor. Of course this would be a major change, so I think its best to leave this method like this right now and improve it in the future.